### PR TITLE
Canonicalizer Unittests, and Break Canonicalization.

### PR DESCRIPTION
**Pull request by Rishav1**
- Addresses CR for #311 
- Adds Canonicalizer unit tests for UnaryOp, Continue and Break.
- Corrected Break syntax test. Added a new test.
- Implemented Break Canonicalizer that moves all the breaks in the loop to the last line within an if condition.

### CODE REVIEW

File: testcases/elichika_tests/canonicalizer/Continue.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        for i in range(10):
+            if i >= 5:
+                continue
+            x += 1
+        """
+        target_code = """
+        x = 0
+        for i in range(10):
+            continued_0 = False
+            if i >= 5:
```
Comment (shinh) [inline testcases/elichika_tests/canonicalizer/Continue.py]: [optional] I'd slightly prefer `==` so the result will be different from the test for `break`. Maybe it'd be also better to change the test for `break` as well?

File: testcases/elichika_tests/canonicalizer/Break.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                breaked_0 = True
+            if not breaked_0:
+                x+=1
+            if breaked_0:
+                break
+        """
+        orig_ast = gast.ast_to_gast(ast.parse(utils.clip_head(orig_code)))
+        target_ast = gast.ast_to_gast(ast.parse(utils.clip_head(target_code)))
+
+        assert compare_ast(self.canonicalizer.visit(orig_ast), target_ast)
```
Comment (shinh) [inline testcases/elichika_tests/canonicalizer/Break.py]: Could you add a few more test cases which test nested/multiple continues?

File: chainer_compiler/elichika/parser/canonicalizer.py [hunk 3]
```diff
@@ -21,41 +21,55 @@ def getflag(self):
...
+            node.body.insert(0, gast.Assign(targets=[gast.Name(id=flag, ctx=gast.Store(), annotation=None)], value=gast.NameConstant(value=False)))
+        breaked_flags = self.for_breaked_stack.pop()
+        bool_values = []
+        for flag in breaked_flags:
+            bool_values.append(gast.Name(id=flag, ctx=gast.Load(), annotation=None))
+        # TODO: Declare the break flags before for loop.
+        if len(bool_values) > 0:
+            if len(bool_values) == 1:
+                cond = bool_values[0]
+            elif len(bool_values > 1):
```
[the remark this answers is not in the document]
Comment (Rishav1) [reply chainer_compiler/elichika/parser/canonicalizer.py]: My bad. This branch wasn't tested. Will fix it.

File: testcases/elichika_tests/canonicalizer/Break.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import unittest
```
[the remark this answers is not in the document]
Comment (Rishav1) [reply testcases/elichika_tests/canonicalizer/Break.py]: I'll add it to travis.

File: testcases/elichika_tests/canonicalizer/Break.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    def setUp(self):
+        self.canonicalizer = canonicalizer.Canonicalizer()
+
+    def test_break(self):
+        orig_code = """
+        x = 0
+        for i in range(10):
+            if i >= 5:
+                break
+            x += 1
```
[the remark this answers is not in the document]
Comment (Rishav1) [reply testcases/elichika_tests/canonicalizer/Break.py]: I agree. In `synax/Break.py`, I used something similar to that. I will correct these tests too.

File: testcases/elichika_tests/canonicalizer/Break.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        x = 0
+        for i in range(10):
+            if i >= 5:
+                breaked_0 = True
+            if not breaked_0:
+                x+=1
+            if breaked_0:
+                break
+        """
+        orig_ast = gast.ast_to_gast(ast.parse(utils.clip_head(orig_code)))
```
[the remark this answers is not in the document]
Comment (Rishav1) [reply testcases/elichika_tests/canonicalizer/Break.py]: This is a nice idea! Will add this.

File: testcases/elichika_tests/canonicalizer/Break.py (hunk 2 above)
Comment (Rishav1) [reply testcases/elichika_tests/canonicalizer/Break.py]: Will do.

**Comment by shinh** [comment]
It looks like this PR is ready to be merged? I'll merge this. If sawada-san has any comments or we find an issue in the current PR, we can handle them in a separate PR. Thanks!

**Pull request merged**